services/device-registry-file/src/main/java/org/eclipse/hono/deviceregistry/file/FileBasedCredentialsService.java

```diff
@@ -652,7 +652,7 @@ private void removeAllForDevice(final String tenantId, final String deviceId, fi
         for (final Object credential : matchingCredentials) {
             final JsonObject credentialsObject = (JsonObject) credential;
             credentialsObject.remove(CredentialsConstants.FIELD_PAYLOAD_DEVICE_ID);
-            removePasswordDetailsFromCredential(credentialsObject);
+            removeConfidentialDataFromCredential(credentialsObject);
```

Comment:
We could reuse `CommonCredential.stripPrivateInfo(...)` instead of `removeConfidentialDataFromCredential(...)` after mapping to `CommonCredential` in the following line.

Reply:
that is an excellent idea :+1: